### ISSUE TITLE
Comparison of two branches is empty

### ISSUE THREAD

**Issue by psiotwo** [b2ihealthcare/snow-owl#477]
I have two branches - one contains a SNOMED CT International, snapshot 2018-07-31, the other SNOMED CT UK extension, snapshot 2019-06-01. The latter is derived from the former. When looking up a SNOMED CT UK concept I correctly find it in the latter and not in the former:

`/snomed-ct/v3/MAIN/2018-07-31/SNOMEDCT-UK-2019-06-01/concepts/1019581000000105`

gives 200, while

`/snomed-ct/v3/MAIN/2018-07-31/concepts/1019581000000105`

gives 404. 

However, when comparing these two branches using `/snomed-ct/v3/compare`
```
{
  "compareBranch": "MAIN/2018-07-31/SNOMEDCT-UK-2019-06-01",
  "baseBranch": "MAIN/2018-07-31",
  "limit" : 1
}
```

I get an empty diff:

```
{
    "baseBranch": "MAIN/2018-07-31/BABYLON",
    "compareBranch": "MAIN/2018-07-31/SNOMEDCT-UK-2019-06-01",
    "compareHeadTimestamp": 1580912379100,
    "newComponents": [],
    "changedComponents": [],
    "deletedComponents": [],
    "totalNew": 0,
    "totalChanged": 0,
    "totalDeleted": 0
}
```

I was expecting to see the UK concepts (e.g. the mentioned 1019581000000105) in the diff. Is it a bug or do I miss something? (tried to consult https://docs.b2i.sg/snow-owl/api/snomed/compare, but maybe I misunderstood the logic of this operation.)

Thanks

Petr

**Comment by cmark** [comment]
Hi @psiotwo,

Thank you for reporting the issue.
This should work pretty much as you have described it, the `1019581000000105` Concept should show up in the diff.
I'll investigate further and get back to you.

Cheers,
Mark

**Comment by cmark** [comment]
Hi @psiotwo,

Could you please retry the compare with the latest Snow Owl 7.5 release without specifying the `limit` value?

Thanks, 
Mark

**Comment by cmark** [comment]
There was a compare bug in the underlying revision compare functionality that in certain cases prevented changes to appear in the compare result. We've fixed that in 7.5.1 release. Resolving as fixed. Feel free to open a new ticket if you still experience the issue.
